src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseImportFromFileService.java

```diff
@@ -85,11 +83,29 @@ public ProgrammingExercise importProgrammingExerciseFromFile(ProgrammingExercise
         if (Boolean.TRUE.equals(programmingExerciseForImport.isStaticCodeAnalysisEnabled())) {
             staticCodeAnalysisService.createDefaultCategories(importedProgrammingExercise);
         }
+        copyEmbeddedFiles(exerciseFilePath.toAbsolutePath().getParent().resolve(FileNameUtils.getBaseName(exerciseFilePath.toString())));
         importRepositoriesFromFile(importedProgrammingExercise, importExerciseDir, oldShortName);
         importedProgrammingExercise.setCourse(course);
+        fileService.scheduleForDirectoryDeletion(importExerciseDir, 5);
```

Comment:
Five minutes until deletion may be a bit short in case the instructor directly changes dates, … of the exercise during the import before saving the new exercise for the first time.

Reply:
If this endpoint is called the instructor has already clicked the import button and the exercise has been generated, so I think 5 minutes are fine.